core/src/main/java/com/linecorp/armeria/client/ClientFactoryBuilder.java

```diff
@@ -419,6 +443,8 @@ public ClientFactoryBuilder tls(KeyManagerFactory keyManagerFactory) {
     @Override
     public ClientFactoryBuilder tlsCustomizer(Consumer<? super SslContextBuilder> tlsCustomizer) {
         requireNonNull(tlsCustomizer, "tlsCustomizer");
+        ensureNoTlsProvider();
+        staticTlsSettingsSet = true;
```

Comment:
Afaik this is the only way to define a custom trustmanager to verify the server, if tlsCustomizer is incompatible with tlsProvider there should be a trustManagerProvider too to configure that (that also makes sense as different routes can need different configurations)

Reply:
A truster manager can be set by overriding `TlsProvider.trustedCertificates()`.

If you need to use `tlsCustomer`, `ClientTlsConfig` is the customization point for `TlsProvider`.